samples/bluetooth/fast_pair/locator_tag/src/motion_detector/platform_thingy.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <math.h>
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/devicetree.h>
+
+#include <bluetooth/services/fast_pair/fmdn.h>
+
+#include "app_motion_detector.h"
+#include "app_ui.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(fp_fmdn, LOG_LEVEL_INF);
+
+/* Minimum required rotation in degrees to consider device as moving */
+#define ROT_THR_DEG		10
+
+/* Minimum required rotation in degrees to consider device as moving */
+#define ROT_THR_RAD		(ROT_THR_DEG * M_PI / 180)
+
+/* Sampling rate (in samples per second) of the motion detection sensor (gyroscope). */
+#define GYRO_SAMPLING_FREQ	10
+
+#define GYRO_NODE	DT_ALIAS(gyro)
+#define GYRO_PWR_NODE	DT_ALIAS(gyro_pwr)
+
+#define GYRO_POLL_THREAD_STACK_SIZE	1024
+#define GYRO_POLL_THREAD_PRIORITY	0
+
+#define SENSOR_VALUE_SET(_var, _int, _frac) \
+	_var.val1 = _int; \
+	_var.val2 = _frac
+
+#define SENSOR_VALUE_SCALE_1000_DEG_PER_SEC_SET(_var)	SENSOR_VALUE_SET(_var, 1000, 0)
```

Comment:
When the scale was set to 125 it was too low. When I turned the Thingy around I was getting something around 200 degrees instead of expected 360 degrees. I didn't observe this issues on scale set to 500, but I think the 1000 is safe value - it's approximately 3 rotations per second which is not impossible.